pkg/kubelet/cm/memorymanager/policy_static.go

```diff
@@ -85,25 +87,28 @@ func (p *staticPolicy) Name() string {
 	return string(policyTypeStatic)
 }
 
-func (p *staticPolicy) Start(s state.State) error {
-	if err := p.validateState(s); err != nil {
-		klog.ErrorS(err, "Invalid state, please drain node and remove policy state file")
+func (p *staticPolicy) Start(ctx context.Context, s state.State) error {
+	logger := klog.FromContext(ctx)
+	if err := p.validateState(logger, s); err != nil {
+		logger.Error(err, "Invalid state, please drain node and remove policy state file")
 		return err
 	}
 	return nil
 }
 
 // Allocate call is idempotent
-func (p *staticPolicy) Allocate(s state.State, pod *v1.Pod, container *v1.Container) (rerr error) {
+func (p *staticPolicy) Allocate(ctx context.Context, s state.State, pod *v1.Pod, container *v1.Container) (rerr error) {
 	// allocate the memory only for guaranteed pods
+	logger := klog.FromContext(ctx)
+	loggerWithValues := klog.LoggerWithValues(logger, "pod", klog.KObj(pod), "containerName", container.Name)
```

Comment:
```suggestion
	logger = klog.LoggerWithValues(logger, "pod", klog.KObj(pod), "containerName", container.Name)
```
It looks like this is commonly used practice in this codebase to just update the logger variable in this kind of cases.

Reply:
Ack, Updated!